index.js

```diff
@@ -1,125 +1,201 @@
-          const subpath = filepath.slice(prefix.length, filepath.length - suffix.length)
-          const pieces = subpath.split(nodePath.sep)
-          const member = pieces.map((name, index) => {
-            const id = identifierfy(name, {
-              prefixReservedWords: pieces.length === 1,
-              prefixInvalidIdentifiers: index === 0
-            })
-            if (id === null) {
-              throw path.buildCodeFrameError(
-                `Could not generate a valid identifier for '${src}'. The '${name}' component could not be converted.`)
-            }
-            return id
-          }).join('$')
-
-          if (lookup[member]) {
-            // hyphen conversion means foo-bar and fooBar will collide.
-            throw path.buildCodeFrameError(`Found colliding members '${member}'`)
-          }
-
-          lookup[member] = src
-          members.push(member)
+        if (hasImportDefaultSpecifier(specifiers)) {
+          throw error('Cannot import the default member')
         }
 
-        if (node.specifiers.length === 0) {
-          path.replaceWithMultiple(members.map(member => t.importDeclaration([], t.stringLiteral(lookup[member]))))
-          return
+        if (!pattern) {
+          throw error(`Missing glob pattern '${pattern}'`)
         }
 
-        const makeImport = (localName, src) => {
-          return t.importDeclaration([t.importDefaultSpecifier(t.identifier(localName))], t.stringLiteral(src))
-        }
-        const makeNamespaceObject = localName => {
-          const properties = members.map(member => {
-            return t.objectProperty(t.identifier(member), t.identifier(`_${localName}_${member}`))
-          })
-          return t.variableDeclaration('const', [t.variableDeclarator(t.identifier(localName), t.objectExpression(properties))])
-        }
-        const freezeNamespaceObject = localName => {
-          return t.expressionStatement(
-            t.callExpression(
-              t.memberExpression(t.identifier('Object'), t.identifier('freeze')),
-              [t.identifier(localName)]))
+        if (!pattern.startsWith('.')) {
+          throw error(`Glob pattern must be relative, was '${pattern}'`)
         }
 
-        const replacement = node.specifiers.map(specifier => {
-          const localName = specifier.local.name
-          if (specifier.type === 'ImportSpecifier') {
-            const importName = specifier.imported.name
-            if (!lookup[importName]) {
-              throw path.buildCodeFrameError(
-                `Could not match import '${importName}' to a module. Available members are '${members.join("', '")}'`)
-            }
+        const cwd = dirname(state.file.opts.filename)
+        const gm = GlobSync(pattern, {cwd, strict: true})
+        const members = generateMembers(gm, cwd)
+        const unique = Object.create(null)
+        members.forEach(m => {
```

Comment:
Sure. This is more out of habit. When using babel the transformation of for/of adds a try/catch, which is useless for arrays. Not using babel anymore though :).